driver/razerchromacommon.c

```diff
@@ -930,6 +930,78 @@ struct razer_report razer_chroma_misc_fn_key_toggle(unsigned char state)
     return report;
 }
 
+/**
+ * Set the keyswitch optimization on the device
+ *
+ * Status Trans Packet Proto DataSize Class CMD Args
+ * 00     1f    0000   00    04       02    02  001400280000               | SET KEY OPTIMIZATION STATE (TYPING SET)
+ * 00     1f    0000   00    05       02    15  010014002800               | SET KEY OPTIMIZATION STATE (TYPING VARSTORE) VARSTORE ????
+ * 00     1f    0000   00    04       02    02  000000000000               | SET KEY OPTIMIZATION STATE (GAMING SET)
+ * 00     1f    0000   00    05       02    15  010000000000               | SET KEY OPTIMIZATION STATE (GAMING VARSTORE) VARSTORE ????
+ */
+struct razer_report razer_chroma_misc_set_keyswitch_optimization(unsigned char variable_storage, unsigned char optimization_mode)
```

Comment:
This function is quite weird, why are you using VARSTORE as parameter two differentiate between two firmware commands on the device? Why not make a separate function that handles e.g. class 0x02 cmd 0x15?

Reply:
It is weird. I believe I started out with two functions before the original PR was submitted, but then tried to consolidate them. I've separated the functions out as requested. I can change the names if something else is preferred. [04d300d](https://github.com/openrazer/openrazer/pull/1759/commits/04d300df0592617ee185d8b2282e0242aab24975)